src/cmd/services/m3dbnode/config/config.go

```diff
@@ -171,14 +190,108 @@ type DBConfiguration struct {
 	Debug config.DebugConfiguration `yaml:"debug"`
 }
 
-// InitDefaultsAndValidate initializes all default values and validates the Configuration.
-// We use this method to validate fields where the validator package falls short.
-func (c *DBConfiguration) InitDefaultsAndValidate() error {
+// LoggingOrDefault returns the logging configuration or defaults.
+func (c *DBConfiguration) LoggingOrDefault() xlog.Configuration {
```

Comment:
Can we just go with `Logging()` and have it default if not set as you have it written? Not sure `XOrDefault()` provides a ton of informational value.

Reply:
Since `DBConfiguration` already has the field of the same name `Logging` we have to have some different name (can't have both field and method name the same), and so `XOrDefault` seems to be the best option for these cases (as opposed to something like `GetLogging()`)

Reply:
Yeah it's mainly because of the name collision we can't make it the same.